auth/shared-signal-receiver/handlers/request-handler.ts

```diff
@@ -30,14 +95,28 @@ export const requestHandler = async (
   jsonBody: unknown,
 ): Promise<APIGatewayProxyResult> => {
   try {
-    for (const { schema, handle } of handlers) {
+    for (const { schema, handle, schemaName, allowedChangeType } of handlers) {
       if (schema.safeParse(jsonBody).success) {
+        if (
+          !isChangeTypeValid(jsonBody as any, schemaName, allowedChangeType)
+        ) {
+          return generateResponse(
+            StatusCodes.BAD_REQUEST,
+            ReasonPhrases.BAD_REQUEST,
+          );
+        }
+        if (!(await isUserValid(jsonBody as any, schemaName))) {
+          return generateResponse(StatusCodes.ACCEPTED, ReasonPhrases.ACCEPTED);
+        }
         return await handle(jsonBody);
       }
     }
 
     return generateResponse(StatusCodes.BAD_REQUEST, ReasonPhrases.BAD_REQUEST);
-  } catch {
+  } catch (error) {
+    if (error instanceof CognitoError) {
+      return generateResponse(StatusCodes.INTERNAL_SERVER_ERROR, error.message);
+    }
     return generateResponse(StatusCodes.BAD_REQUEST, ReasonPhrases.BAD_REQUEST);
   }
```

Comment:
This `try/catch` can be removed since if we bubble-up a `CognitoError` this already produces the correct error response in the main handler.